crates/sui-json-rpc/src/threshold_bls_api.rs

```diff
@@ -0,0 +1,142 @@
+use sui_json_rpc_types::{
+    SuiCertifiedTransactionEffects, SuiTBlsSignObjectCommitmentType,
+    SuiTBlsSignRandomnessObjectResponse,
+};
+use sui_open_rpc::Module;
+use sui_types::base_types::ObjectID;
+use sui_types::crypto::construct_tbls_randomness_object_message;
+use sui_types::object::{Object, ObjectRead};
+
+pub struct ThresholdBlsApiImpl {
+    state: Arc<AuthorityState>,
+}
+
+impl ThresholdBlsApiImpl {
+    pub fn new(state: Arc<AuthorityState>) -> Self {
+        Self { state }
+    }
+
+    /// Check that the given layout represents a Randomness object.
+    fn is_randomness_object(layout: &MoveStructLayout) -> bool {
+        let MoveStructLayout::WithTypes{type_, fields: _} = layout else { return false; };
+        let prefix = "0000000000000000000000000000000000000002::randomness::Randomness";
+        type_.to_canonical_string().starts_with(prefix)
+    }
+
+    /// Get the object and check if it is a Randomness object.
+    async fn get_randomness_object(&self, object_id: ObjectID) -> RpcResult<Object> {
+        let obj_read = self
+            .state
+            .get_object_read(&object_id)
+            .await
+            .map_err(|e| anyhow!(e))?;
+        let ObjectRead::Exists(_obj_ref, obj, layout) = obj_read else {
+            Err(anyhow!("Object not found"))? };
+        let Some(layout) = layout else { Err(anyhow!("Object does not have a layout"))? };
+        if !Self::is_randomness_object(&layout) {
+            Err(anyhow!("Not a Randomness object"))?
+        }
+        Ok(obj)
+    }
+
+    /// Return true if the given object exists according to my local view.
+    ///
+    /// Currently only checks if the object exists in the local storage, but in the future
+    /// validators will verify that the object had been created in a transaction that was committed.
+    async fn verify_object_alive_and_committed(&self, object_id: ObjectID) -> RpcResult<()> {
+        let _obj = self.get_randomness_object(object_id).await?;
+        Ok(())
+    }
+
+    async fn verify_effects_cert(
+        &self,
+        object_id: ObjectID,
+        effects_cert: &SuiCertifiedTransactionEffects,
+    ) -> RpcResult<()> {
+        if effects_cert.auth_sign_info.epoch != self.state.epoch() {
+            Err(anyhow!(
+                "Old effects certificate, check instead if committed by consensus"
+            ))?
+        }
+        // Check the certificate.
+        let _committee = self
+            .state
+            .committee_store()
+            .get_committee(&self.state.epoch())
+            .map_err(|e| anyhow!(e))?
+            .ok_or_else(|| anyhow!("Committee not available"))?;
+
+        // TODO: convert SuiTransactionEffects to TransactionEffects before the next line.
+        // effects_cert
+        //     .auth_sign_info
+        //     .verify(&effects_cert.effects, &committee)
+        //     .map_err(|e| anyhow!(e))?;
```

Comment:
do we still need this todo?

Reply:
yes, i will implement that in a follow up PR